include/ap/argument_parser.hpp

```diff
@@ -1474,77 +1507,36 @@ class argument_parser {
         }
     }
 
-    /// @brief Structure representing a command-line argument.
-    struct cmd_argument {
-        enum class type_discriminator : bool { flag, value };
-
-        cmd_argument() = default;
-        cmd_argument(const cmd_argument&) = default;
-        cmd_argument(cmd_argument&&) = default;
-        cmd_argument& operator=(const cmd_argument&) = default;
-
-        /**
-         * @brief Constructor of a command-line argument.
-         * @param discriminator Type discriminator (flag or value).
-         * @param value The value of the argument.
-         */
-        cmd_argument(const type_discriminator discriminator, const std::string& value)
-        : discriminator(discriminator), value(value) {}
-
-        ~cmd_argument() = default;
-
-        /**
-         * @brief Equality operator for comparing cmd_argument instances.
-         * @param other Another cmd_argument to compare with.
-         * @return Boolean statement of equality comparison.
-         */
-        inline bool operator==(const cmd_argument& other) const {
-            return this->discriminator == other.discriminator and this->value == other.value;
-        }
-
-        type_discriminator discriminator;
-        std::string value;
-    };
-
-    using cmd_argument_list = std::vector<cmd_argument>;
-    using cmd_argument_list_iterator = typename cmd_argument_list::const_iterator;
-
-    using argument_ptr_type = std::unique_ptr<argument::detail::argument_interface>;
-    using argument_opt_type = std::optional<std::reference_wrapper<argument::detail::argument_interface>>;
-    using argument_list_type = std::vector<argument_ptr_type>;
-    using argument_list_iterator_type = typename argument_list_type::iterator;
-    using argument_list_const_iterator_type = typename argument_list_type::const_iterator;
-    using argument_predicate_type = std::function<bool(const argument_ptr_type&)>;
-
     /**
-     * @brief Function to create a predicate for finding arguments by name.
+     * @brief Function to create a predicate for finding arguments by thename.
      * @param name The name of the argument.
      * @return Argument predicate based on the provided name.
      */
-    [[nodiscard]] inline argument_predicate_type _name_eq_predicate(const std::string_view& name) const {
+    [[nodiscard]] argument_predicate_type _name_eq_predicate(const std::string_view& name) const {
```

Comment:
replace sepparate _name_eq_predicate functions for the primary or (primary and secondary) names with a single function which takes the `ap::argument::detail::argument_name` as parameter